### ISSUE TITLE
insserv error when installing on Ubuntu 16.04.1

### ISSUE THREAD

**Issue by braykov** [docker/docker#26469]
```
sudo apt-get install docker-engine 
[sudo] password for user_name: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed
  docker-engine
0 to upgrade, 1 to newly install, 0 to remove and 0 not to upgrade.
Need to get 0 B/19,5 MB of archives.
After this operation, 102 MB of additional disk space will be used.
Selecting previously unselected package docker-engine.
(Reading database ... 319751 files and directories currently installed.)
Preparing to unpack .../docker-engine_1.12.1-0~xenial_amd64.deb ...
Unpacking docker-engine (1.12.1-0~xenial) ...
Processing triggers for man-db (2.7.5-1) ...
Processing triggers for ureadahead (0.100.0-19) ...
Processing triggers for systemd (229-4ubuntu7) ...
Setting up docker-engine (1.12.1-0~xenial) ...
insserv: warning: script is corrupt or invalid: /etc/init.d/../rc2.d/S01shm_load
Processing triggers for systemd (229-4ubuntu7) ...
Processing triggers for ureadahead (0.100.0-19) ...
```

Please note the line `insserv: warning: script is corrupt or invalid: /etc/init.d/../rc2.d/S01shm_load`
This file does not exist under Ubuntu. Actually `/etc/init.d` contains no sub-folders.

**Output of `docker version`:**

```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
```

**Output of `docker info`:**

```
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 2
Server Version: 1.12.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 6
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-36-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.33 GiB
Name: leno
ID: YF5Z:V4M4:BYEG:POEF:FTN6:VEVU:5WU4:VAVM:U4OM:X6ZN:5HW2:5PP3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical

**Steps to reproduce the issue:**
1. purge previous installation: `sudo apt-get purge docker-engine`
2. install the latest version

**Describe the results you received:**

Several iterations of install reinstall both report the *insserv* problem

**Describe the results you expected:**

Expected no installation errors

**Comment by braykov** [comment]
Verifying I am pulling from the right repo:

```
apt-cache policy docker-engine
docker-engine:
  Installed: 1.12.1-0~xenial
  Candidate: 1.12.1-0~xenial
  Version table:
 *** 1.12.1-0~xenial 500
        500 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
     1.12.0-0~xenial 500
        500 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
     1.11.2-0~xenial 500
        500 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
     1.11.1-0~xenial 500
        500 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
     1.11.0-0~xenial 500
        500 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
```

**Comment by braykov** [comment]
And now the problem shows up again when following the installation instructions and reaching the step for configuring Docker to start on boot:

```
sudo systemctl enable docker
Synchronizing state of docker.service with SysV init with /lib/systemd/systemd-sysv-install...
Executing /lib/systemd/systemd-sysv-install enable docker
insserv: warning: script is corrupt or invalid: /etc/init.d/../rc2.d/S01shm_load
insserv: warning: script is corrupt or invalid: /etc/init.d/../rc2.d/S01shm_load
```

**Comment by mlaventure** [comment]
@braykov insserv doesn't come installed with 16.04 as it uses systemd instead.

Do you know how it got into your system? After double checking, running `curl -sSL https://get.docker.com | sh` did not add `insserv` on my digital ocean droplet.

**Comment by braykov** [comment]
@mlaventure, no I don't know. But my Ubuntu is not a clean install. I have had several upgrades up to 16.04.1. 
Also I am not using the curl way of installing. I am following the instructions with adding a PPA and using APT to install docker-engine. AFAIK this is the preferred way.
And now the real problem is that the daemon is not starting automatically on boot (as I mentioned above).

**Comment by mlaventure** [comment]
@braykov the script basically just add the new source for you automatically.

As for the failure, I can't find a logical explanation why installing docker would break your `S01shm_load` init script (we don't provide such a script).

Could you share its content and from which package it is coming from (i.e. `dpkg -S /etc/init.d/../rc2.d/S01shm_load`)?

**Comment by braykov** [comment]
I will repeat this from my first post above:

This file does not exist under Ubuntu. Actually `/etc/init.d` contains no sub-folders.

**Comment by mlaventure** [comment]
@braykov the actual path is `/etc/rc2.d/S01shm_load`)

**Comment by braykov** [comment]
@mlaventure OMG, got it:\

```
#!/bin/sh
case "$1" in
start) mkdir /var/lock/subsys 2>/dev/null
touch /var/lock/subsys/listener
rm /dev/shm 2>/dev/null
mkdir /dev/shm 2>/dev/null
mount -t tmpfs shmfs -o size=4096m /dev/shm ;;
*) echo error
exit 1 ;;
esac
```

**Comment by mlaventure** [comment]
Could you tell us where it's coming from: `dpkg -S /etc/rc2.d/S01shm_load`

And to be sure can you run `/sbin/init --help` or similar to assert that your system is indeed using systemd.

**Comment by braykov** [comment]
```
dpkg-query: no path found matching pattern /etc/rc2.d/S01shm_load
```
Does this mean I can delete this file?


... and now `/sbin/init --help`:
```
init [OPTIONS...] {COMMAND}

Send control commands to the init daemon.

     --help      Show this help
     --no-wall   Don't send wall message before halt/power-off/reboot

Commands:
  0              Power-off the machine
  6              Reboot the machine
  2, 3, 4, 5     Start runlevelX.target unit
  1, s, S        Enter rescue mode
  q, Q           Reload init daemon configuration
  u, U           Reexecute init daemon
```

**Comment by mlaventure** [comment]
@braykov this I cannot tell you :sweat_smile: 

You can first try to move it somewhere safe and see if it affects your system or not.

**Comment by braykov** [comment]
OK, the strangest thing happened. I went to `mv` the file with a backup extension and it said there is no such file. It really has disappeared. I have no idea why and how, but it must have been in the last half an hour. 
I purged docker-engine, did apt update and upgrade, then installed docker-engine again. 
It did not report any error. So I restarted to make sure the daemon starts on reboot - it does. :-)
Should I close this now?
